// @azure/ai-text-analytics/dist-esm/src/lro/poller.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
import { __awaiter } from "tslib";
import { delay } from "@azure/core-http";
import { Poller } from "@azure/core-lro";
/**
 * Common properties and methods of analysis Pollers.
 */
export class AnalysisPoller extends Poller {
    constructor() {
        super(...arguments);
        /**
         * Defines how much time the poller is going to wait before making a new request to the service.
         */
        this.updateIntervalInMs = 2000;
    }
    /**
     * The method used by the poller to wait before attempting to update its operation.
     */
    delay() {
        return __awaiter(this, void 0, void 0, function* () {
            return delay(this.updateIntervalInMs);
        });
    }
}
/**
 * Common properties and methods of polling operations.
 */
export class AnalysisPollOperation {
    constructor(state) {
        this.state = state;
    }
    /**
     * @summary Serializes the Poller operation.
     */
    toString() {
        return JSON.stringify({
            state: this.state
        });
    }
}
